crates/next-dev/benches/util/mod.rs

```diff
@@ -131,105 +132,131 @@ pub fn resume_on_error<F: FnOnce() + UnwindSafe>(f: F) {
     }
 }
 
-pub trait AsyncBencherExtension {
-    fn try_iter_async<I, O, S, SF, W, WF, R, F, T, TF>(
+pub trait AsyncBencherExtension<A: AsyncExecutor> {
+    fn try_iter_async<I, O, S, SF, R, F, U, UF, T, TF>(
```

Comment:
This method is getting insanely verbose. As part of this PR I don't think we need the `warmup` fn any longer. I also think we could eliminate `O` by ensuring that our routine always returns `I` (or takes `&mut I` and returns nothing, but I couldn't manage to make that work with ownership previously).

Reply:
We can't return `I` from the routine since that would require closing the page as part of the measurement.

Reply:
We make the routine to capture its own timing to make that possible.

Reply:
Similar to what I did in https://github.com/vercel/turbo/pull/2700 then?